app/controllers/webhook_events_controller.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+class WebhookEventsController < ApplicationController
+  skip_before_action :verify_authenticity_token
+  skip_before_action :authenticate_user!
+
+  before_action :verify_payload_signature
+  before_action :verify_organization_presence
+  before_action :verify_sender_presence
+
+  def create
+    if respond_to? event_handler
+      send event_handler
+      head :ok, content_type: 'application/json'
+    else
+      not_found
+    end
+  end
+
+  def handle_ping
+    return if @organization.is_webhook_active?
+    @organization.update_attributes(is_webhook_active: true)
+  end
+
+  private
+
+  def event_name
+    request.headers['X-GitHub-Event']
+  end
+
+  def event_handler
+    @event_handler ||= "handle_#{event_name}".to_sym
+  end
+
+  def verify_payload_signature
+    return unless Rails.application.secrets.webhook_secret.present?
+
+    algorithm, signature = request.headers['X-Hub-Signature'].split('=')
+    payload_validated = actual_payload_signature(algorithm) == signature
```

Comment:
@cyhsutw I'd like to make a few small changes here. This is pretty much what you have, except we do a [`Rack::Utils.secure_compare`](http://www.rubydoc.info/github/rack/rack/Rack/Utils.secure_compare) instead of `==`.

I'd also like to send back statuses which gives GitHub a heads up that something is wrong 😁 

```ruby
  def receive_and_verify_payload
    payload_body = request.body.read
    render json: { message: 'No payload received' }, status: 400 if payload_body.nil?

    expected_hmac = OpenSSL::HMAC.hexdigest(OpenSSL::Digest.new('sha1'), Rails.application.secrets.webhook_secret, payload_body)

    expected_signature = "sha1=#{expected_hmac}"
    received_signature = request.env['HTTP_X_HUB_SIGNATURE']

    unless received_signature && Rack::Utils.secure_compare(received_signature, expected_signature)
      render json: { message: 'Invalid payload signature' }, status: :forbidden
    end
  end
```